File: src/hoi4utils/clausewitz_coding/units/Regiment.java
package hoi4utils.clausewitz_coding.units;

public class Regiment {
	enum subunitType {
		COMBAT_BATALLION,
		SUPPORT_COMPANY
	}

	// etc.

	int x;      // position in template
	int y;      // position in template

	public Regiment(int x, int y) {
		this.x = x;
		this.y = y;
	}
}

File: src/main/scala/com/hoi4utils/hoi4/units/Regiment.scala
package com.hoi4utils.hoi4.units

/** The kind of subunit in a regiment. */
enum SubunitType:
  case CombatBattalion, SupportCompany

/**
 * A single regiment in a division template, positioned at (x,y). 
 * @param x  horizontal slot in the template
 * @param y  vertical slot in the template
 */
case class Regiment(x: Int, y: Int)